### PR TITLE
Generate wasm parts

### PR DESCRIPTION
**Pull request by kpob**

### CODE REVIEW

**Review by kubaplas — APPROVED** [review]
@zie1ony check macros part

File: odra-macros/src/ast/deployer_utils.rs [hunk 1]
```diff
@@ -104,7 +104,7 @@ impl TryFrom<&'_ ModuleIR> for NewContractExpr {
         let env_ident = utils::ident::env();
         let args = module
             .constructor()
-            .map(|f| ref_utils::runtime_args_block(&f))
+            .map(|f| fn_utils::runtime_args_block(&f, ref_utils::insert_arg_stmt))
```
Comment (zie1ony) [inline odra-macros/src/ast/deployer_utils.rs:107]: Looks like a nice abstraction. Hopefully it's not gonna bite.

File: odra-macros/src/ir/mod.rs [hunk 2]
```diff
@@ -149,6 +149,17 @@ impl ModuleIR {
         ))
     }
 
+    pub fn wasm_parts_mod_ident(&self) -> Result<syn::Ident, syn::Error> {
+        let module_ident = self.module_ident()?;
```
Comment (zie1ony) [inline odra-macros/src/ir/mod.rs]: Maybe `self.module_ident_snake_case()`?

File: odra-macros/src/ast/wasm_parts_utils.rs [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    utils
+};
+use syn::parse_quote;
+
+pub fn param_parameters(func: &FnIR) -> syn::Expr {
+    let params = func
+        .named_args()
+        .iter()
+        .map(|arg| arg.name_and_ty())
+        .filter_map(|result| match result {
```
Comment (zie1ony) [inline odra-macros/src/ast/wasm_parts_utils.rs]: `result.ok()` is shorter.

**Review by zie1ony — APPROVED** [review]

**Pull request merged**